### ISSUE TITLE
ty should report `@runtime_checkable` on non-protocols as an error

### ISSUE THREAD

**Issue by A5rocks** [astral-sh/ty#4470]
### Summary

See title. Here's a repro:

```py
from typing import runtime_checkable

@runtime_checkable
class Bweh: ...
```

**Comment by AlexWaygood** [comment]
Ah, right, this fails at runtime:

```pytb
  File "main.py", line 3, in <module>
    @runtime_checkable
     ^^^^^^^^^^^^^^^^^
  File "/lib/python314.zip/typing.py", line 2278, in runtime_checkable
    raise TypeError('@runtime_checkable can be only applied to protocol classes,'
                    ' got %r' % cls)
TypeError: @runtime_checkable can be only applied to protocol classes, got <class 'Bweh'>
```
